[CoordinatedGraphics] Introduce CoordinatedPlatformLayer

carlosgcampos opened the pull request:
#### e009963a7b178a306760b645e6a12e6a75ca4f4b
<pre>
[CoordinatedGraphics] Introduce CoordinatedPlatformLayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=283966">https://bugs.webkit.org/show_bug.cgi?id=283966</a>

Reviewed by Alejandro G. Castro.

Rename CoordinatedGraphicsLayer as GraphicsLayerCoordinated since it&apos;s
the coordinated graphics implementation of GraphicsLayer. Also add
CoordinatedPlatformLayer class to keep the layers composition state to
be applied by the threaded compositor. This is an initial patch that
still uses NicosiaCompositionLayer in CoordinatedPlatformLayer so that
it doesn&apos;t require changes in the composition side.

* Source/WebCore/page/scrolling/nicosia/ScrollingStateNodeNicosia.cpp:
(WebCore::LayerRepresentation::platformLayerFromGraphicsLayer):
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeNicosia.cpp:
(WebCore::ScrollingTreeNicosia::applyLayerPositionsInternal):
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/Damage.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::supportsContentsTiling):
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::isGraphicsLayerCoordinated const):
(WebCore::GraphicsLayer::isCoordinatedGraphicsLayer const): Deleted.
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h:
(Nicosia::PlatformLayer::createUpdateScope):
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
(Nicosia::SceneIntegration::createUpdateScope):
(Nicosia::SceneIntegration::UpdateScope::UpdateScope):
(Nicosia::SceneIntegration::UpdateScope::~UpdateScope):
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.h:
* Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.cpp:
(WebCore::SkiaPaintingEngine::recordDisplayList const):
(WebCore::SkiaPaintingEngine::paintIntoGraphicsContext const):
(WebCore::SkiaPaintingEngine::paintGraphicsLayerIntoBuffer const):
(WebCore::SkiaPaintingEngine::createBuffer const):
(WebCore::SkiaPaintingEngine::paintLayer):
(WebCore::SkiaPaintingEngine::postPaintingTask):
(WebCore::SkiaPaintingEngine::performPaintingTask):
* Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.h:
* Source/WebCore/platform/graphics/texmap/GraphicsLayerAsyncContentsDisplayDelegateTextureMapper.cpp:
(WebCore::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper):
(WebCore::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper::updateGraphicsLayer):
* Source/WebCore/platform/graphics/texmap/GraphicsLayerAsyncContentsDisplayDelegateTextureMapper.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedAnimatedBackingStoreClient.cpp:
(WebCore::CoordinatedAnimatedBackingStoreClient::create):
(WebCore::CoordinatedAnimatedBackingStoreClient::CoordinatedAnimatedBackingStoreClient):
(WebCore::CoordinatedAnimatedBackingStoreClient::update):
(WebCore::CoordinatedAnimatedBackingStoreClient::requestBackingStoreUpdateIfNeeded):
(WebCore::CoordinatedAnimatedBackingStoreClient::setCoverRect): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedAnimatedBackingStoreClient.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp:
(WebCore::CoordinatedBackingStoreProxy::updateIfNeeded):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerCairo.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.cpp: Added.
(WebCore::CoordinatedPlatformLayer::create):
(WebCore::CoordinatedPlatformLayer::CoordinatedPlatformLayer):
(WebCore::CoordinatedPlatformLayer::setOwner):
(WebCore::CoordinatedPlatformLayer::owner const):
(WebCore::CoordinatedPlatformLayer::invalidateClient):
(WebCore::CoordinatedPlatformLayer::notifyCompositionRequired):
(WebCore::CoordinatedPlatformLayer::setPosition):
(WebCore::CoordinatedPlatformLayer::syncPosition):
(WebCore::CoordinatedPlatformLayer::position const):
(WebCore::CoordinatedPlatformLayer::setBoundsOrigin):
(WebCore::CoordinatedPlatformLayer::syncBoundsOrigin):
(WebCore::CoordinatedPlatformLayer::boundsOrigin const):
(WebCore::CoordinatedPlatformLayer::setAnchorPoint):
(WebCore::CoordinatedPlatformLayer::anchorPoint const):
(WebCore::CoordinatedPlatformLayer::setSize):
(WebCore::CoordinatedPlatformLayer::size const):
(WebCore::CoordinatedPlatformLayer::setTransform):
(WebCore::CoordinatedPlatformLayer::transform const):
(WebCore::CoordinatedPlatformLayer::setChildrenTransform):
(WebCore::CoordinatedPlatformLayer::childrenTransform const):
(WebCore::CoordinatedPlatformLayer::didUpdateLayerTransform):
(WebCore::CoordinatedPlatformLayer::setVisibleRect):
(WebCore::CoordinatedPlatformLayer::setTransformedVisibleRect):
(WebCore::CoordinatedPlatformLayer::setScrollingNodeID):
(WebCore::CoordinatedPlatformLayer::setDrawsContent):
(WebCore::CoordinatedPlatformLayer::setMasksToBounds):
(WebCore::CoordinatedPlatformLayer::setPreserves3D):
(WebCore::CoordinatedPlatformLayer::setBackfaceVisibility):
(WebCore::CoordinatedPlatformLayer::setOpacity):
(WebCore::CoordinatedPlatformLayer::setContentsVisible):
(WebCore::CoordinatedPlatformLayer::contentsVisible const):
(WebCore::CoordinatedPlatformLayer::setContentsOpaque):
(WebCore::CoordinatedPlatformLayer::setContentsRect):
(WebCore::CoordinatedPlatformLayer::setContentsRectClipsDescendants):
(WebCore::CoordinatedPlatformLayer::setContentsClippingRect):
(WebCore::CoordinatedPlatformLayer::setContentsScale):
(WebCore::CoordinatedPlatformLayer::setContentsBuffer):
(WebCore::CoordinatedPlatformLayer::setContentsBufferNeedsDisplay):
(WebCore::CoordinatedPlatformLayer::setContentsImage):
(WebCore::CoordinatedPlatformLayer::setContentsColor):
(WebCore::CoordinatedPlatformLayer::setContentsTileSize):
(WebCore::CoordinatedPlatformLayer::setContentsTilePhase):
(WebCore::CoordinatedPlatformLayer::setDirtyRegion):
(WebCore::CoordinatedPlatformLayer::setDamage):
(WebCore::CoordinatedPlatformLayer::setFilters):
(WebCore::CoordinatedPlatformLayer::setMask):
(WebCore::CoordinatedPlatformLayer::setReplica):
(WebCore::CoordinatedPlatformLayer::setBackdrop):
(WebCore::CoordinatedPlatformLayer::setBackdropRect):
(WebCore::CoordinatedPlatformLayer::setAnimations):
(WebCore::CoordinatedPlatformLayer::setChildren):
(WebCore::CoordinatedPlatformLayer::setEventRegion):
(WebCore::CoordinatedPlatformLayer::setDebugBorder):
(WebCore::CoordinatedPlatformLayer::setShowRepaintCounter):
(WebCore::CoordinatedPlatformLayer::needsBackingStore const):
(WebCore::CoordinatedPlatformLayer::updateBackingStore):
(WebCore::CoordinatedPlatformLayer::updateContents):
(WebCore::CoordinatedPlatformLayer::purgeBackingStores):
(WebCore::CoordinatedPlatformLayer::paint):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.h: Added.
(WebCore::CoordinatedPlatformLayer::id const):
(WebCore::CoordinatedPlatformLayer::compositionLayer const):
(WebCore::CoordinatedPlatformLayer::lock):
(WebCore::CoordinatedPlatformLayer::client const):
(WebCore::CoordinatedPlatformLayer::hasPendingTilesCreation const):
(WebCore::CoordinatedPlatformLayer::hasImageBackingStore const):
(WebCore::CoordinatedPlatformLayer::WTF_GUARDED_BY_LOCK):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp: Added.
(WebCore::GraphicsLayer::create):
(WebCore::GraphicsLayerCoordinated::GraphicsLayerCoordinated):
(WebCore::GraphicsLayerCoordinated::~GraphicsLayerCoordinated):
(WebCore::GraphicsLayerCoordinated::primaryLayerID const):
(WebCore::GraphicsLayerCoordinated::setNeedsDisplay):
(WebCore::GraphicsLayerCoordinated::setNeedsDisplayInRect):
(WebCore::GraphicsLayerCoordinated::markDamageRectsUnreliable):
(WebCore::GraphicsLayerCoordinated::setPosition):
(WebCore::GraphicsLayerCoordinated::syncPosition):
(WebCore::GraphicsLayerCoordinated::setBoundsOrigin):
(WebCore::GraphicsLayerCoordinated::syncBoundsOrigin):
(WebCore::GraphicsLayerCoordinated::setAnchorPoint):
(WebCore::GraphicsLayerCoordinated::setSize):
(WebCore::GraphicsLayerCoordinated::setScrollingNodeID):
(WebCore::GraphicsLayerCoordinated::setTransform):
(WebCore::GraphicsLayerCoordinated::setChildrenTransform):
(WebCore::GraphicsLayerCoordinated::setDrawsContent):
(WebCore::GraphicsLayerCoordinated::setMasksToBounds):
(WebCore::GraphicsLayerCoordinated::setPreserves3D):
(WebCore::GraphicsLayerCoordinated::setBackfaceVisibility):
(WebCore::GraphicsLayerCoordinated::setOpacity):
(WebCore::GraphicsLayerCoordinated::setContentsVisible):
(WebCore::GraphicsLayerCoordinated::setContentsOpaque):
(WebCore::GraphicsLayerCoordinated::setContentsRect):
(WebCore::GraphicsLayerCoordinated::setContentsRectClipsDescendants):
(WebCore::GraphicsLayerCoordinated::setContentsTileSize):
(WebCore::GraphicsLayerCoordinated::setContentsTilePhase):
(WebCore::GraphicsLayerCoordinated::setContentsClippingRect):
(WebCore::GraphicsLayerCoordinated::setContentsNeedsDisplay):
(WebCore::GraphicsLayerCoordinated::setContentsToPlatformLayer):
(WebCore::GraphicsLayerCoordinated::setContentsDisplayDelegate):
(WebCore::GraphicsLayerCoordinated::createAsyncContentsDisplayDelegate):
(WebCore::GraphicsLayerCoordinated::setContentsToImage):
(WebCore::GraphicsLayerCoordinated::setContentsToSolidColor):
(WebCore::GraphicsLayerCoordinated::usesContentsLayer const):
(WebCore::GraphicsLayerCoordinated::setChildren):
(WebCore::GraphicsLayerCoordinated::addChild):
(WebCore::GraphicsLayerCoordinated::addChildAtIndex):
(WebCore::GraphicsLayerCoordinated::addChildBelow):
(WebCore::GraphicsLayerCoordinated::addChildAbove):
(WebCore::GraphicsLayerCoordinated::replaceChild):
(WebCore::GraphicsLayerCoordinated::willModifyChildren):
(WebCore::GraphicsLayerCoordinated::setEventRegion):
(WebCore::GraphicsLayerCoordinated::deviceOrPageScaleFactorChanged):
(WebCore::GraphicsLayerCoordinated::setFilters):
(WebCore::GraphicsLayerCoordinated::setMaskLayer):
(WebCore::GraphicsLayerCoordinated::setReplicatedByLayer):
(WebCore::GraphicsLayerCoordinated::setBackdropFilters):
(WebCore::GraphicsLayerCoordinated::setBackdropFiltersRect):
(WebCore::GraphicsLayerCoordinated::addAnimation):
(WebCore::GraphicsLayerCoordinated::removeAnimation):
(WebCore::GraphicsLayerCoordinated::pauseAnimation):
(WebCore::GraphicsLayerCoordinated::suspendAnimations):
(WebCore::GraphicsLayerCoordinated::resumeAnimations):
(WebCore::GraphicsLayerCoordinated::isRunningTransformAnimation const):
(WebCore::GraphicsLayerCoordinated::transformRelatedPropertyDidChange):
(WebCore::GraphicsLayerCoordinated::acceleratedAnimationsForTesting const):
(WebCore::GraphicsLayerCoordinated::setShowDebugBorder):
(WebCore::GraphicsLayerCoordinated::setShowRepaintCounter):
(WebCore::dumpInnerLayer):
(WebCore::GraphicsLayerCoordinated::dumpAdditionalProperties const):
(WebCore::GraphicsLayerCoordinated::filtersCanBeComposited const):
(WebCore::GraphicsLayerCoordinated::noteLayerPropertyChanged):
(WebCore::GraphicsLayerCoordinated::setNeedsUpdateLayerTransform):
(WebCore::GraphicsLayerCoordinated::flushCompositingState):
(WebCore::GraphicsLayerCoordinated::flushCompositingStateForThisLayerOnly):
(WebCore::GraphicsLayerCoordinated::computePositionRelativeToBase const):
(WebCore::GraphicsLayerCoordinated::computePixelAlignmentIfNeeded):
(WebCore::GraphicsLayerCoordinated::updateGeometry):
(WebCore::GraphicsLayerCoordinated::computeLayerTransformIfNeeded):
(WebCore::GraphicsLayerCoordinated::clampToSizeIfRectIsInfinite):
(WebCore::GraphicsLayerCoordinated::updateVisibleRect):
(WebCore::GraphicsLayerCoordinated::updateDamage):
(WebCore::GraphicsLayerCoordinated::updateDirtyRegion):
(WebCore::GraphicsLayerCoordinated::updateBackdropFilters):
(WebCore::GraphicsLayerCoordinated::updateBackdropFiltersRect):
(WebCore::GraphicsLayerCoordinated::updateAnimations):
(WebCore::GraphicsLayerCoordinated::updateIndicators):
(WebCore::GraphicsLayerCoordinated::updateContents):
(WebCore::GraphicsLayerCoordinated::commitLayerChanges):
(WebCore::GraphicsLayerCoordinated::needsCommit const):
(WebCore::GraphicsLayerCoordinated::recursiveCommitChanges):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.h: Added.
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::scrollbarInclusionForVisibleRect):
(WebCore::RenderLayerCompositor::visibleRectForLayerFlushing const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::notifyCompositionRequired):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::layerForTransientZoom const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:

Canonical link: <a href="https://commits.webkit.org/287731@main">https://commits.webkit.org/287731@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aa73bf4766127693145189c1fb8c00ee0269d580

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80668 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59675 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/34602 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85192 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31650 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82779 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68736 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7984 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/63000 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20794 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83737 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/53132 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/73432 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43305 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/50413 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/27591 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30109 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71564 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28115 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86625 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7894 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/5568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/71294 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/8069 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69267 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/70536 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14554 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/13505 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12495 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7856 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/13375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7695 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/11214 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/9500 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great refactor!

carlosgcampos committed:
[CoordinatedGraphics] Introduce CoordinatedPlatformLayer
https://bugs.webkit.org/show_bug.cgi?id=283966

Reviewed by Alejandro G. Castro.

Rename CoordinatedGraphicsLayer as GraphicsLayerCoordinated since it's
the coordinated graphics implementation of GraphicsLayer. Also add
CoordinatedPlatformLayer class to keep the layers composition state to
be applied by the threaded compositor. This is an initial patch that
still uses NicosiaCompositionLayer in CoordinatedPlatformLayer so that
it doesn't require changes in the composition side.

* Source/WebCore/page/scrolling/nicosia/ScrollingStateNodeNicosia.cpp:
(WebCore::LayerRepresentation::platformLayerFromGraphicsLayer):
* Source/WebCore/page/scrolling/nicosia/ScrollingTreeNicosia.cpp:
(WebCore::ScrollingTreeNicosia::applyLayerPositionsInternal):
* Source/WebCore/platform/TextureMapper.cmake:
* Source/WebCore/platform/graphics/Damage.h:
* Source/WebCore/platform/graphics/GraphicsLayer.cpp:
(WebCore::GraphicsLayer::supportsContentsTiling):
* Source/WebCore/platform/graphics/GraphicsLayer.h:
(WebCore::GraphicsLayer::isGraphicsLayerCoordinated const):
(WebCore::GraphicsLayer::isCoordinatedGraphicsLayer const): Deleted.
* Source/WebCore/platform/graphics/nicosia/NicosiaPlatformLayer.h:
(Nicosia::PlatformLayer::createUpdateScope):
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.cpp:
(Nicosia::SceneIntegration::createUpdateScope):
(Nicosia::SceneIntegration::UpdateScope::UpdateScope):
(Nicosia::SceneIntegration::UpdateScope::~UpdateScope):
* Source/WebCore/platform/graphics/nicosia/NicosiaSceneIntegration.h:
* Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.cpp:
(WebCore::SkiaPaintingEngine::recordDisplayList const):
(WebCore::SkiaPaintingEngine::paintIntoGraphicsContext const):
(WebCore::SkiaPaintingEngine::paintGraphicsLayerIntoBuffer const):
(WebCore::SkiaPaintingEngine::createBuffer const):
(WebCore::SkiaPaintingEngine::paintLayer):
(WebCore::SkiaPaintingEngine::postPaintingTask):
(WebCore::SkiaPaintingEngine::performPaintingTask):
* Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.h:
* Source/WebCore/platform/graphics/texmap/GraphicsLayerAsyncContentsDisplayDelegateTextureMapper.cpp:
(WebCore::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper):
(WebCore::GraphicsLayerAsyncContentsDisplayDelegateTextureMapper::updateGraphicsLayer):
* Source/WebCore/platform/graphics/texmap/GraphicsLayerAsyncContentsDisplayDelegateTextureMapper.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedAnimatedBackingStoreClient.cpp:
(WebCore::CoordinatedAnimatedBackingStoreClient::create):
(WebCore::CoordinatedAnimatedBackingStoreClient::CoordinatedAnimatedBackingStoreClient):
(WebCore::CoordinatedAnimatedBackingStoreClient::update):
(WebCore::CoordinatedAnimatedBackingStoreClient::requestBackingStoreUpdateIfNeeded):
(WebCore::CoordinatedAnimatedBackingStoreClient::setCoverRect): Deleted.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedAnimatedBackingStoreClient.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp:
(WebCore::CoordinatedBackingStoreProxy::updateIfNeeded):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.h:
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.h: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerCairo.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp: Removed.
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.cpp: Added.
(WebCore::CoordinatedPlatformLayer::create):
(WebCore::CoordinatedPlatformLayer::CoordinatedPlatformLayer):
(WebCore::CoordinatedPlatformLayer::setOwner):
(WebCore::CoordinatedPlatformLayer::owner const):
(WebCore::CoordinatedPlatformLayer::invalidateClient):
(WebCore::CoordinatedPlatformLayer::notifyCompositionRequired):
(WebCore::CoordinatedPlatformLayer::setPosition):
(WebCore::CoordinatedPlatformLayer::syncPosition):
(WebCore::CoordinatedPlatformLayer::position const):
(WebCore::CoordinatedPlatformLayer::setBoundsOrigin):
(WebCore::CoordinatedPlatformLayer::syncBoundsOrigin):
(WebCore::CoordinatedPlatformLayer::boundsOrigin const):
(WebCore::CoordinatedPlatformLayer::setAnchorPoint):
(WebCore::CoordinatedPlatformLayer::anchorPoint const):
(WebCore::CoordinatedPlatformLayer::setSize):
(WebCore::CoordinatedPlatformLayer::size const):
(WebCore::CoordinatedPlatformLayer::setTransform):
(WebCore::CoordinatedPlatformLayer::transform const):
(WebCore::CoordinatedPlatformLayer::setChildrenTransform):
(WebCore::CoordinatedPlatformLayer::childrenTransform const):
(WebCore::CoordinatedPlatformLayer::didUpdateLayerTransform):
(WebCore::CoordinatedPlatformLayer::setVisibleRect):
(WebCore::CoordinatedPlatformLayer::setTransformedVisibleRect):
(WebCore::CoordinatedPlatformLayer::setScrollingNodeID):
(WebCore::CoordinatedPlatformLayer::setDrawsContent):
(WebCore::CoordinatedPlatformLayer::setMasksToBounds):
(WebCore::CoordinatedPlatformLayer::setPreserves3D):
(WebCore::CoordinatedPlatformLayer::setBackfaceVisibility):
(WebCore::CoordinatedPlatformLayer::setOpacity):
(WebCore::CoordinatedPlatformLayer::setContentsVisible):
(WebCore::CoordinatedPlatformLayer::contentsVisible const):
(WebCore::CoordinatedPlatformLayer::setContentsOpaque):
(WebCore::CoordinatedPlatformLayer::setContentsRect):
(WebCore::CoordinatedPlatformLayer::setContentsRectClipsDescendants):
(WebCore::CoordinatedPlatformLayer::setContentsClippingRect):
(WebCore::CoordinatedPlatformLayer::setContentsScale):
(WebCore::CoordinatedPlatformLayer::setContentsBuffer):
(WebCore::CoordinatedPlatformLayer::setContentsBufferNeedsDisplay):
(WebCore::CoordinatedPlatformLayer::setContentsImage):
(WebCore::CoordinatedPlatformLayer::setContentsColor):
(WebCore::CoordinatedPlatformLayer::setContentsTileSize):
(WebCore::CoordinatedPlatformLayer::setContentsTilePhase):
(WebCore::CoordinatedPlatformLayer::setDirtyRegion):
(WebCore::CoordinatedPlatformLayer::setDamage):
(WebCore::CoordinatedPlatformLayer::setFilters):
(WebCore::CoordinatedPlatformLayer::setMask):
(WebCore::CoordinatedPlatformLayer::setReplica):
(WebCore::CoordinatedPlatformLayer::setBackdrop):
(WebCore::CoordinatedPlatformLayer::setBackdropRect):
(WebCore::CoordinatedPlatformLayer::setAnimations):
(WebCore::CoordinatedPlatformLayer::setChildren):
(WebCore::CoordinatedPlatformLayer::setEventRegion):
(WebCore::CoordinatedPlatformLayer::setDebugBorder):
(WebCore::CoordinatedPlatformLayer::setShowRepaintCounter):
(WebCore::CoordinatedPlatformLayer::needsBackingStore const):
(WebCore::CoordinatedPlatformLayer::updateBackingStore):
(WebCore::CoordinatedPlatformLayer::updateContents):
(WebCore::CoordinatedPlatformLayer::purgeBackingStores):
(WebCore::CoordinatedPlatformLayer::paint):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.h: Added.
(WebCore::CoordinatedPlatformLayer::id const):
(WebCore::CoordinatedPlatformLayer::compositionLayer const):
(WebCore::CoordinatedPlatformLayer::lock):
(WebCore::CoordinatedPlatformLayer::client const):
(WebCore::CoordinatedPlatformLayer::hasPendingTilesCreation const):
(WebCore::CoordinatedPlatformLayer::hasImageBackingStore const):
(WebCore::CoordinatedPlatformLayer::WTF_GUARDED_BY_LOCK):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp: Added.
(WebCore::GraphicsLayer::create):
(WebCore::GraphicsLayerCoordinated::GraphicsLayerCoordinated):
(WebCore::GraphicsLayerCoordinated::~GraphicsLayerCoordinated):
(WebCore::GraphicsLayerCoordinated::primaryLayerID const):
(WebCore::GraphicsLayerCoordinated::setNeedsDisplay):
(WebCore::GraphicsLayerCoordinated::setNeedsDisplayInRect):
(WebCore::GraphicsLayerCoordinated::markDamageRectsUnreliable):
(WebCore::GraphicsLayerCoordinated::setPosition):
(WebCore::GraphicsLayerCoordinated::syncPosition):
(WebCore::GraphicsLayerCoordinated::setBoundsOrigin):
(WebCore::GraphicsLayerCoordinated::syncBoundsOrigin):
(WebCore::GraphicsLayerCoordinated::setAnchorPoint):
(WebCore::GraphicsLayerCoordinated::setSize):
(WebCore::GraphicsLayerCoordinated::setScrollingNodeID):
(WebCore::GraphicsLayerCoordinated::setTransform):
(WebCore::GraphicsLayerCoordinated::setChildrenTransform):
(WebCore::GraphicsLayerCoordinated::setDrawsContent):
(WebCore::GraphicsLayerCoordinated::setMasksToBounds):
(WebCore::GraphicsLayerCoordinated::setPreserves3D):
(WebCore::GraphicsLayerCoordinated::setBackfaceVisibility):
(WebCore::GraphicsLayerCoordinated::setOpacity):
(WebCore::GraphicsLayerCoordinated::setContentsVisible):
(WebCore::GraphicsLayerCoordinated::setContentsOpaque):
(WebCore::GraphicsLayerCoordinated::setContentsRect):
(WebCore::GraphicsLayerCoordinated::setContentsRectClipsDescendants):
(WebCore::GraphicsLayerCoordinated::setContentsTileSize):
(WebCore::GraphicsLayerCoordinated::setContentsTilePhase):
(WebCore::GraphicsLayerCoordinated::setContentsClippingRect):
(WebCore::GraphicsLayerCoordinated::setContentsNeedsDisplay):
(WebCore::GraphicsLayerCoordinated::setContentsToPlatformLayer):
(WebCore::GraphicsLayerCoordinated::setContentsDisplayDelegate):
(WebCore::GraphicsLayerCoordinated::createAsyncContentsDisplayDelegate):
(WebCore::GraphicsLayerCoordinated::setContentsToImage):
(WebCore::GraphicsLayerCoordinated::setContentsToSolidColor):
(WebCore::GraphicsLayerCoordinated::usesContentsLayer const):
(WebCore::GraphicsLayerCoordinated::setChildren):
(WebCore::GraphicsLayerCoordinated::addChild):
(WebCore::GraphicsLayerCoordinated::addChildAtIndex):
(WebCore::GraphicsLayerCoordinated::addChildBelow):
(WebCore::GraphicsLayerCoordinated::addChildAbove):
(WebCore::GraphicsLayerCoordinated::replaceChild):
(WebCore::GraphicsLayerCoordinated::willModifyChildren):
(WebCore::GraphicsLayerCoordinated::setEventRegion):
(WebCore::GraphicsLayerCoordinated::deviceOrPageScaleFactorChanged):
(WebCore::GraphicsLayerCoordinated::setFilters):
(WebCore::GraphicsLayerCoordinated::setMaskLayer):
(WebCore::GraphicsLayerCoordinated::setReplicatedByLayer):
(WebCore::GraphicsLayerCoordinated::setBackdropFilters):
(WebCore::GraphicsLayerCoordinated::setBackdropFiltersRect):
(WebCore::GraphicsLayerCoordinated::addAnimation):
(WebCore::GraphicsLayerCoordinated::removeAnimation):
(WebCore::GraphicsLayerCoordinated::pauseAnimation):
(WebCore::GraphicsLayerCoordinated::suspendAnimations):
(WebCore::GraphicsLayerCoordinated::resumeAnimations):
(WebCore::GraphicsLayerCoordinated::isRunningTransformAnimation const):
(WebCore::GraphicsLayerCoordinated::transformRelatedPropertyDidChange):
(WebCore::GraphicsLayerCoordinated::acceleratedAnimationsForTesting const):
(WebCore::GraphicsLayerCoordinated::setShowDebugBorder):
(WebCore::GraphicsLayerCoordinated::setShowRepaintCounter):
(WebCore::dumpInnerLayer):
(WebCore::GraphicsLayerCoordinated::dumpAdditionalProperties const):
(WebCore::GraphicsLayerCoordinated::filtersCanBeComposited const):
(WebCore::GraphicsLayerCoordinated::noteLayerPropertyChanged):
(WebCore::GraphicsLayerCoordinated::setNeedsUpdateLayerTransform):
(WebCore::GraphicsLayerCoordinated::flushCompositingState):
(WebCore::GraphicsLayerCoordinated::flushCompositingStateForThisLayerOnly):
(WebCore::GraphicsLayerCoordinated::computePositionRelativeToBase const):
(WebCore::GraphicsLayerCoordinated::computePixelAlignmentIfNeeded):
(WebCore::GraphicsLayerCoordinated::updateGeometry):
(WebCore::GraphicsLayerCoordinated::computeLayerTransformIfNeeded):
(WebCore::GraphicsLayerCoordinated::clampToSizeIfRectIsInfinite):
(WebCore::GraphicsLayerCoordinated::updateVisibleRect):
(WebCore::GraphicsLayerCoordinated::updateDamage):
(WebCore::GraphicsLayerCoordinated::updateDirtyRegion):
(WebCore::GraphicsLayerCoordinated::updateBackdropFilters):
(WebCore::GraphicsLayerCoordinated::updateBackdropFiltersRect):
(WebCore::GraphicsLayerCoordinated::updateAnimations):
(WebCore::GraphicsLayerCoordinated::updateIndicators):
(WebCore::GraphicsLayerCoordinated::updateContents):
(WebCore::GraphicsLayerCoordinated::commitLayerChanges):
(WebCore::GraphicsLayerCoordinated::needsCommit const):
(WebCore::GraphicsLayerCoordinated::recursiveCommitChanges):
* Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.h: Added.
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::scrollbarInclusionForVisibleRect):
(WebCore::RenderLayerCompositor::visibleRectForLayerFlushing const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::notifyCompositionRequired):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::layerForTransientZoom const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:

Canonical link: https://commits.webkit.org/287731@main
